### DISCUSSION TITLE
v4 search Filter webpart customization, Handlebars customizations

### DISCUSSION THREAD

**Discussion by fonbrauzen** [microsoft-search/pnp-modern-search#2946] [Ideas]
**Is your feature request related to a problem? Please describe.**
search filters web part doesn't have Handlebars customizers

**Describe the solution you'd like**
add via Search results to search filters web part ability to use Handlebars customizers

**Describe alternatives you've considered**
The only customization for search filters currently - web component, which is a bit of an overkill (in this case)

**Additional context**
In my case I have the array of numbers, which I want to sort 1, 2, 3, 10 and not like in JS 1, 10, 2, 3.
To do this I need a small  JS function:
```JS
let a = [1 , 10000, 10, 20 ];
a.sort(); //default, incorrect => [1, 10, 10000, 20]
a.sort((a, b) => { return a - b; }) //with sort function, => [1, 10, 20, 10000]      TODO as HandleBars customization
```
According to your docs, currently, it's not possible.

**Comment by wobba** [comment]
Did you try something like `{{#each (sortBy data.items "DocId") as |item|}}`? You don't say how your array is coming into the template. Also take a look at https://github.com/helpers/handlebars-helpers#sort

**Comment by fonbrauzen** [comment]
@wobba  the main problem is - sorting for numbers in JS, so basically Handlebars "sort" isn't working, because JS sorting the array wrong (see above). That's why make sense to expose Handlebars customizations everywhere, where HTML templating is possible (this way would be possible to insert in the template any JS function you need). 
"You don't say how your array is coming into the template." - I have an HTML template for "search filters" webpart, there 1 of refiners is array of int32.

**Reply by wobba** [reply]
Sounds like a good PR we'd accept :)

**Comment by fonbrauzen** [comment]
@wobba  if it was an invitation to contribute, for me it would take some time to make it. For someone knowing the project probably not that much time, maybe together with SPFX 1.17.1 upgrade (if you following the SPFX releases with the new PnP search version...)

**Reply by wobba** [reply]
Every code base takes some time to get into for sure. And I plan to bump SPFx as well.
